cpp/src/parquet/bloom_filter_benchmark.cc

```diff
@@ -0,0 +1,148 @@
+  std::vector<T> values(1024);
+  std::vector<uint8_t> heap;
+  test::GenerateData(1024, values.data(), &heap);
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(
+      BlockSplitBloomFilter::OptimalNumOfBytes(1024, /*fpp=*/0.05));
+  BloomFilter* filter = &block_split_bloom_filter;
+  ::benchmark::DoNotOptimize(filter);
+  std::vector<uint64_t> hashes(1024);
+  for (auto _ : state) {
+    if constexpr (std::is_same_v<DType, FLBAType>) {
+      filter->Hashes(values.data(), test::kGenerateDataFLBALength,
+                     static_cast<int>(values.size()), hashes.data());
+    } else {
+      filter->Hashes(values.data(), static_cast<int>(values.size()), hashes.data());
+    }
+  }
+  state.SetItemsProcessed(state.iterations() * values.size());
+}
+
+static void BM_InsertHash(::benchmark::State& state) {
+  using T = int32_t;
+  std::vector<T> values(1024);
+  std::vector<uint8_t> heap;
+  test::GenerateData(1024, values.data(), &heap);
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(
+      BlockSplitBloomFilter::OptimalNumOfBytes(1024, /*fpp=*/0.05));
+  BloomFilter* filter = &block_split_bloom_filter;
+  ::benchmark::DoNotOptimize(filter);
+  std::vector<uint64_t> hashes(1024);
+  filter->Hashes(values.data(), static_cast<int>(values.size()), hashes.data());
+  for (auto _ : state) {
+    for (auto hash : hashes) {
+      filter->InsertHash(hash);
+    }
+  }
+  state.SetItemsProcessed(state.iterations() * values.size());
+}
+
+static void BM_BatchInsertHash(::benchmark::State& state) {
+  using T = int32_t;
+  std::vector<T> values(1024);
+  std::vector<uint8_t> heap;
+  test::GenerateData(1024, values.data(), &heap);
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(
+      BlockSplitBloomFilter::OptimalNumOfBytes(1024, /*fpp=*/0.05));
+  BloomFilter* filter = &block_split_bloom_filter;
+  ::benchmark::DoNotOptimize(filter);
+  std::vector<uint64_t> hashes(1024);
+  filter->Hashes(values.data(), static_cast<int>(values.size()), hashes.data());
+  for (auto _ : state) {
+    filter->InsertHashes(hashes.data(), static_cast<int>(hashes.size()));
+  }
+  state.SetItemsProcessed(state.iterations() * values.size());
+}
+
+static void BM_FindHash(::benchmark::State& state) {
+  using T = int32_t;
+  std::vector<T> values(1024);
+  std::vector<uint8_t> heap;
+  test::GenerateData(1024, values.data(), &heap);
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(
+      BlockSplitBloomFilter::OptimalNumOfBytes(1024, /*fpp=*/0.05));
+  BloomFilter* filter = &block_split_bloom_filter;
+  ::benchmark::DoNotOptimize(filter);
+  std::vector<uint64_t> hashes(1024);
+  filter->Hashes(values.data(), static_cast<int>(values.size()), hashes.data());
+  filter->InsertHashes(hashes.data(), static_cast<int>(hashes.size()));
+  for (auto _ : state) {
+    for (auto hash : hashes) {
+      filter->FindHash(hash);
+    }
```

Comment:
This measures performance for existing hash values, but can you also add a benchmark for missing hash values? The performance should be different.

Reply:
Okay, will do it